lecture_5/homework/app/services/reports/siege_report.rb

```diff
@@ -3,10 +3,28 @@
 module Reports
   class SiegeReport
     def initialize(building:)
+      @building = building
+      @warriors = building.warriors
     end
 
     def call
-      raise NotImprementedYet
+      siege = @warriors.present? ? @building.siege_ability = granary_ability : @building.siege_ability = 0
```

Comment:
To sprawdzenie spokojnie by się mogło znaleźć gdzieś "wyżej", żeby nie obciążać logiki serwisu :) Serwis najlepiej jeśli ma jedną odpowiedzialność, a Twój serwis poza liczeniem ile dni przetrwa dany budynek, stwierdza upadłość budynku jeśli nie ma w nim wojowników :)